### PR TITLE
chore: improve dev server performance

### PR DESCRIPTION
**Pull request by huntabyte**
This PR is a bit hefty, but takes care of a lot of chores, some of which severely impact the dev server performance and make the project tough to contribute to if you're not running an H100.

### Icon Import Strategy
We were importing all icons from the index `lucide-svelte`, which caused Vite to optimize the thousands of icons. Fortunately, @AdrianGonz97's PR was merged https://github.com/lucide-icons/lucide/pull/1707 and we can now import the icons directly:
```ts
import Check from 'lucide-svelte/icons/check'
```
Vite is only going to optimize that single `Check` component instead of their entire collection.

To provide a similar DX to what we had before, all the icons being used are now being exported from `src/docs/components/icons/index.ts`

```ts
// src/docs/components/icons.index.ts

export { default as Check } from 'lucide-svelte/icons/check'
export { default as X } from 'lucide-svelte/icons/x'
// ...
```

The icons can be consumed in the docs using the `$icons` import alias:

```svelte
<script lang="ts">
	import { Check, X } from '$icons/index.js'
</script>
```

So we still get the benefits of Vite's optimization, but only for the icons we actually use.

### Code Highlighting

Although `shikiji` has been merged back in with [shiki](https://shiki.style), the plugin we use for our nice code blocks, [rehype-pretty-code](https://rehype-pretty-code.netlify.app/) won't have support for regular `shiki` until it's out of beta (see: https://github.com/rehype-pretty/rehype-pretty-code/issues/171).

In the meantime, we'll use the latest version of `shikiji`, which is leaps and bounds more performant and optimized than the one we're currently using. Once `shiki` is out of beta, it will be a matter of replacing the dependencies and won't be a big lift. Once this gets merged we can create an issue to remind me to update once the time comes.

Both the Rehype Plugin for preprocessing the markdown as well as the cached preview component highlighters have been updated to this.

### SvelteKit 2 & Vite 5

The project has been updated to SvelteKit 2 & Vite 5, with all other dependencies (mostly testing-related) updated alongside it. Vite 5 not only comes with a bunch of improvements, but additionally support for a stable release of Vitest (v1). The entire test suite now takes less than 10 seconds to run (on my machine), which is pretty nice!

### Minor Tweaks

Made a few optimizations around the gathering of the builder data and caching the highlighted code during dev. 

## Results

Running on a Mac Studio, node_modules & .svelte-kit directories are removed before each video/cache is cleared.

### Before

https://github.com/melt-ui/melt-ui/assets/64506580/95e35e92-f0f9-4f2d-8e32-eef51633241f

### After

https://github.com/melt-ui/melt-ui/assets/64506580/8c416f3a-5ec2-473d-98b2-8c9524aef5f1

### CODE REVIEW

**Comment by TGlide** [comment]
i fucking love you

File: mdsvex.config.js [hunk 1]
```diff
@@ -8,31 +9,47 @@ import { toHtml } from 'hast-util-to-html';
...
 
 const __dirname = fileURLToPath(new URL('.', import.meta.url));
 
+/** @type {import('rehype-pretty-code').Options} */
 const prettyCodeOptions = {
 	theme: 'github-dark',
 	keepBackground: false,
 	onVisitLine(node) {
 		if (node.children.length === 0) {
+			console.log('empty line');
```
Comment (TGlide) [inline mdsvex.config.js:29]: leftover?

**Review by TGlide — APPROVED** [review]

File: mdsvex.config.js (hunk 1 above)
Comment (huntabyte) [reply mdsvex.config.js:29]: Yessir!

**Comment by huntabyte** [comment]
Hey @TGlide, just added a route `/repro` which will redirect users to a StackBlitz with the Melt UI playground repo (which I submitted a PR to update).

So now users just have to head to https://melt-ui.com/repro to easily spin up a reproduction environment :) would have done this in a separate PR but merge conflicts cause SK v2 so sneaking it in here 😈

**Pull request merged**